packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-header/pipeline-actions.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import type { ConnectedProps } from 'react-redux';
+import {
+  Button,
+  css,
+  cx,
+  spacing,
+  Link,
+  Icon,
+} from '@mongodb-js/compass-components';
+
+import type { RootState } from '../../../modules';
+import { runAggregation } from '../../../modules/aggregation';
+import { changeWorkspace } from '../../../modules/workspace';
+import { toggleOptions } from '../../../modules/options';
+
+const containerStyles = css({
+  display: 'flex',
+});
+
+const defaultMargins = css({
+  marginLeft: spacing[1],
+  marginRight: spacing[1],
+});
```

Comment:
If you use `:not(:first-child)` and just required left margin instead you will not be left with weird margins that affect anything outside of this section. You can also use `gap` CSS property on the container and then your components don't need to know anything about their margins at all